templates/components/engines/python/agent/tools/__init__.py

```diff
@@ -19,15 +19,6 @@ class ToolFactory:
         ToolType.LOCAL: "app.engine.tools",
     }
 
-    @staticmethod
-    @cached(
-        LRUCache(maxsize=100),
-        key=lambda tool_type, tool_name, config: (
-            tool_type,
-            tool_name,
-            json.dumps(config, sort_keys=True),
-        ),
-    )
```

Comment:
We added this to reuse the tool instance for the OpenAPI tool feature. But it's also applied to all the other tools too. I think we now should accept the issue and find another way to reuse the parsed openapi result later. @marcusschiesser what do you think?

Reply:
how about caching it in the OpenAPI tool (see my TS code changes for OpenAPI)?

Reply:
@leehuwuj while doing this (as you're going to change the OpenAPI tool), you can also add the OpenAPI spec description to the tool that is retrieving the spec (see https://github.com/run-llama/create-llama/blob/3079162806017ae7de88f98a9e013066160ca111/templates/components/engines/typescript/agent/tools/openapi-action.ts#L134) - then both OpenAPI tools would behave the same in TS and Python

Reply:
> how about caching it in the OpenAPI tool (see my TS code changes for OpenAPI)?

I updated to cache the OpenAPI tool with the same mechanism as TS.  


> you can also add the OpenAPI spec description to the tool that is retrieving the spec

Unfortunately, we just simply combine the OpenAPI and Request tool, they're both ToolSpec so the tool descriptions are automatically inferenced by the function description. To add the openapi info into the tool description we must change how ToolSpec behaves which makes the code become complicated.

Reply:
Got it. added a new task for this